### ISSUE TITLE
Broken JWT builder: version bumps of mercure and php JWT dependency

### ISSUE THREAD

**Issue by teutat3s** [mvanduijker/laravel-mercure-broadcaster#6]
Used versions for testing:
mercure: v0.10
laravel-mercure-broadcaster: v3.1.0

- [version bump jwt dep](https://github.com/lcobucci/jwt/releases/tag/3.4.0)
-> new deprecation notice causes exception in laravel

Possible fixes:
- version freeze jwt dep?
https://github.com/lcobucci/jwt/releases/tag/3.3.3

- [version bump mercure incoming](https://github.com/dunglas/mercure/releases/tag/v0.11.0-beta.1)?

Error messages:
- in Browser:
```
HTTP 401 unauthorized
```

- mercure logs:
```
{"level":"info","ts":1606474124.5431101,"logger":"http.handlers.mercure","msg":"Topic selectors not matched or not provided","remote_addr":"x.x.x.x:46442","error":"unable to parse JWT: signature is invalid"}
```

- php laravel logs:
```
trigger_error('Implicit conversion of keys from strings is deprecated. Please use InMemory or LocalFileReference classes.', E_USER_DEPRECATED);
```

---------
More upgrades incoming:
Upgrading docs for incoming php jwt dep:
https://github.com/lcobucci/jwt/blob/master/docs/upgrading.md

**Comment by mvanduijker** [comment]
Thanks for the bug report. Will try to take a look this evening.

**Comment by mvanduijker** [comment]
I fixed the error when using version 3.4 of https://github.com/lcobucci/jwt

I noticed that:
* Mercure 0.11 doesn't seem to work.
* version 4.0 for https://github.com/lcobucci/jwt has changed it's api a lot and is also needed to support php 8. These changes should go in a new major version.

Could you check if master works for you now on Mercure 0.10?

**Comment by dunglas** [comment]
> Mercure 0.11 doesn't seem to work.

Be sure to read the upgrade guide: https://mercure.rocks/docs/hub/UPGRADE
You must either upgrade the configuration to use the `Caddyfile` or use the *legacy* build.

**Comment by mvanduijker** [comment]
@dunglas I'm trying to get the legacy build to work in my integration test. But it's outputting the following error: `unexpected signing method` Still have to figure out what I'm doing wrong.

**Comment by mvanduijker** [comment]
Should be fixed from version 3.1.1
